packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/direct-edit/DiagramDirectEditInput.tsx

```diff
@@ -77,10 +77,10 @@ export const DiagramDirectEditInput = ({ labelId, editingKey, onClose, transform
     newLabel: initialLabel,
   });
 
-  const { addErrorMessage, addMessages } = useMultiToast();
+  const cancelled = useRef<boolean>(false);
```

Comment:
Why are you using a `useRef` instead of putting this in the state?

Reply:
I tried several approaches, the first one being obviously to add `cancelled` into the state, but the way `onBlur` is invoked, it never sees the state change.

I'm not really satisfied with `useRef`, but it's the only way I found to make it work.